code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/LegacyNotificationBuilderTest.kt

```diff
@@ -0,0 +1,257 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.DEFAULT_CHANNEL_ID
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.SILENT_CHANNEL_NAME
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import org.junit.Assert.assertArrayEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+import kotlin.test.assertNull
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class LegacyNotificationBuilderTest {
+
+    @Mock
+    private lateinit var pushTemplate: BasicPushTemplate
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var context: Context
+    private val SUT = LegacyNotificationBuilder
+
+    @Before
+    fun setUp() {
```

Comment:
IMO creating tests that ultimately check if a notification has been constructed(or not) as desired with all the values present/absent would be more robust as opposed to individually asserting on a single setter at a time. I feel verifying individual keys will make it cumbersome to test interrelated  keys/settings later on.

I am wondering if we have the following setup 
-  a map/bundle with all the keys available for the template
- an extension method on this map/bundle that allows removing/unsetting a list of values/keys.
- an extension method on this map/bundle that allows overriding a list of keys via an input map

it will enable us to write tests in the format of 

```
val myTemplateMap = getLegacyTemplateDataMap()  // returns fully populated template with all test data
                              .without(list(KEY1, KEY7,....)) // some keys with additional clauses/defaults in source code
                              .overwrite(mapOf(.....))
val mapData = MapData(myTemplateMap)
LegacyNotificationBuilder.construct(..., mapData,....)
verifyConstruction(mapData, notification, omittedKeys, overwrittenKeys)  // intelligent method to compare at notification data and built notification
```

@PratthamArora , @spoorthipujariadobe , @PravinPK - Thoughts?

Reply:
Agree with your suggestion. This PR has the utility methods you're referring to #9. Should be simpler to use those to check if the notification was constructed properly for given data. You can add/remove keys or replace values to test the various code branches we have. Since these are unit tests, we can adopt a white box approach

Reply:
sure, will create a mock data class with dummy data and merge all the happy cases for the construct method